### DISCUSSION TITLE
UMinho - Bibliografia não gerada

### DISCUSSION THREAD

**Discussion by joaomariocosta** [joaomlourenco/novathesis#209] [Q&A] [answered]
Olá,

estou a tentar utilizar a formatação da UMinho - eeg. 
No entanto, a página da bibliografia nunca é gerada. 
Já tentei colocar o ficheiro de teste "bibliography" noutras directorias, mas sem sucesso.
Reparei tambem que no ficheiro "..\novathesis\NOVAthesisFiles\Schools\uminho\univ-defaults.ldf" aparece comentado "%\ntsetprintorder{mainmatter}{
 %  chapters,        % Print document chapters
  % bib,             % Print the bibliography.  Change to
                    %    “bib[title=⟨title⟩]” to redefine the title!
  % appendices,      % Print appendices, if any!
   %annexes,         % Print annexes, if any!
 %}" 
 
mas imagino que não seja dai o problema, sendo que os chapters, os apendices e os anexos são gerados correctamente.

Segunda pergunta, para utilizar o estilo "Chicago, autor data" qual será o comando que se tem de utilizar? 

com os melhores cumprimentos,
João Costa

**Comment by b-pereira** [comment]
Boas João,

1. Experimentaste compilar o template sem modificações e ver se gera a bibliografia? Entretanto, tenta compilar 3 vezes (o Tex gera ficheiros auxiliares para a biliografia, e por vezes, é preciso fazer isto nas primeiras vezes).
2.  A bibliografia (onde tens as tua "base de dados" de referências) fica aqui:
![imagem](https://user-images.githubusercontent.com/944511/126309459-4f1fffc5-8fa8-4440-839e-85f331aa18ae.png)
3. A configuração do tipo de citações e do tipo de biliografia é aqui:
![imagem](https://user-images.githubusercontent.com/944511/126309604-1d610cb3-3ad2-44d9-9eb1-ce2c366b5725.png)
Aquilo que pretende (author-year) pode ser alcançado com o `APA6` ou `authoryear-comp`. No ficheiro da diretoria da imagem acima tire o comentário da linha 48 ou 52 conforme.
![imagem](https://user-images.githubusercontent.com/944511/126310055-ef6e693e-3374-49f0-8847-3dc346903096.png)

Se o APA ou esse author-year não corresponder ao que pretende existe o `chicago-authordate`, mas preciso testar para ver se funciona. Se quiser experimentar, seria o que estaria no linha 48, só que com `chicago-authordate`.

Para as citações tipo APA, porque tem o `natbib=true`. Não sei se o no `authoryear-comp` dá para fazer o mesmo. Tinha que testar.
![imagem](https://user-images.githubusercontent.com/944511/126311203-4f15d56c-9aba-45d0-9234-936ab2609fd3.png)

Na imagem acima a linha:
```latex
\cite[see][chap 2]{latexcompanion}
```
tem que ser no português se quiser usar em português:

```latex
\cite[ver][cap. 2]{latexcompanion}
```

**Reply by joaomariocosta** [reply]
Olá,

sim, sem nenhuma modificação do meu lado, a bibliografia não é gerada.

Em relação ao ponto 3 (estando a trabalhar com o MacTex) verifiquei no "TeX Live Utility" que tenho instalado o package do "biblatex-chicago" instalado. Tentei importar/utilizar o package de várias formas que fui encontrando no 02_biblatex.tex e no 7_packages.tex, mas sem sucesso (inclusive o da linha 48 e 52 do 02_biblatex) com erros de compilação.

Mas terei primeiro de por a gerar a bibliografia para tentar ver como ficam as referencias e as notas de rodapé.

**Reply by b-pereira** [reply]
Que versão do template estás a usar?

**Reply by b-pereira** [reply]
Para já não podes usar o `biblatex-chicago`. Este é um _package_ independente e terias de invocar
```latex
\usepackage[notes,backend=biber]{biblatex-chicago}
```
Ou seja estarias a tentar redefinir o _package_ `biblatex`. Isto teria que ser adicionado como funcionalidade para humanidades.

Para ver o que pretendes vê este documento [The Chicago Citation Style with biblatex](https://www.overleaf.com/latex/examples/the-chicago-citation-style-with-biblatex/pdqqrmwtdqpc). Está esta a usar apenas notas de rodapé, porque é o que esse package usa com `\autocite[pagina]{author9999}` para cada nota de rodapé, o `\cite{author9999}` coloca-te autor, nome da obra, data o que acho que não é o que tu queres. **O importante é que esse está a usar o `biblatex-chicago`.**

Além disso, o que eu disse é para usares o **_style_** **`chicago-authordate`**. Estive a experiementar e não me dá bons resultados. 

Ou seja o que te disse era a melhor alternativa: ou usas o `apa6` direto como está nessa linha, ou usas o `authoryear-comp` e se quiseres até podes adicionar o `natbib=true` e p `backend=biber` e poder usar a versões: (Autor, Data) - `citep`; Autor, Data - `cite`; Autot _et al_ - `\citeauthor`.

Se quiseres fazer como está no documento acima e ter notas em rodapé das citações podes usar o `\footcite` - este apresenta uma versão sintética com Autor, Data-, ou podes usar o `\fullfootcite` se quiseres toda a informação da referência bibliográfica.

Sobre a bibliografia não aparecer, vê se não apagaste os ficheiros dentro da pasta `Chapters`, porque para a bibliografia aparecer tem haver citações. É só um palpite. Se não envia MP para o meu Facebook  https://www.facebook.com/pereirabruno05/ que eu tento resolver isso.

**Reply by joaomariocosta** [reply]
Olá de novo,

já desisti do estilo chicago, pq não estava a conseguir importar os packages nem coisa do genero. Já gastei tempo de mais a tentar por isso a dar 😂

Quanto a bibliografia, eu já fiz reset do master, tou com 0 alterações feitas e continuo sem ver a bibliografia gerada.

**Comment by joaomlourenco** [comment, marked as answer]
Tens que utilizar o "biber" e não o "bibtex" para gerar a bibliografia.   Não sei que editor estás a utilizar, mas normalmente o processador de bibliografia pré-configurado é o bibtex.  Tens que alterar para o biber.  Pede ajuda ao google se necessário.

**Reply by b-pereira** [reply]
@joaomlourenco parece que o @joaomariocosta conseguiu resolver o problema. Ele está a usar o _MacTex_, onde a instalação completa vem com o _TexShop_. A bibliografia não era gerada porque a forma como o editor ou @joaomariocosta estavam a trabalhar:
1. Não usava o `biber`; 
2. Havia problemas no _workflow_ do João em que os passos `xelatex -> biber -> xelatex -> xelatex` não eram efetuados;  Eu não estando familiarizado com o Mactex e o TexShop, nem tendo MacOS foi um pouco difícil de perceber, mas ele lá encontrou a solução.
Aliás, enviou-me o erro:
```
Package biblatex Warning: Please (re)run Biber on the file:
(biblatex)                template
(biblatex)                and rerun LaTeX afterwards.
 )
 ```
Como disse o  problema do _workflow_  foi resolvido e ele já tem bibliografia.

Para referência futura [Bibtex, Latex compiling](https://tex.stackexchange.com/questions/204291/bibtex-latex-compiling) onde tem a maioria dos editores e a configuração do `biber`. @joaomlourenco, poderia isto deveria estar em qualquer lado na wiki, como uma FAQ? (ou já existe uma?).

**Reply by joaomariocosta** [reply]
Um dos problemas tambem é que estava inicialmente a compilar utilizando o LuaLaTex e não obtinha erro nos logs, simplesmente não gerava a bibliografia.
Depois de alterar para xelatex por sugestão do @b-pereira notei esse erro e depois acabei por resolver. 

obrigado pelas respostas e pela ajuda 😄
